### DISCUSSION TITLE
Add a config setting to include the RAW request body to Logs

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#9941] [Ideas] [answered]
More and more CP requests are done via Axios rather than your older method - and I use more Axios, the "downside" of this, is the `$_POST` array in CP requests are mostly empty. When clients face issues/system exceptions it's harder to debug them without having the post params ( we encountered an issue with your Craft search where a client got an exception - wrong SQL Syntax - that I cannot reproduce and seeing the search string would be useful)

Of course we can extend your Craft Logger and always overwrite the `getContextMessage` function (what I currently do. except for this project) but I wonder if it might be useful to add a config var whether the RAW body should be included directly in your `craft\base\LogTargetTrait` to include `file_get_contents('php://input')`

**Comment by brandonkelly** [comment, marked as answer]
Good idea. Just swapped `$_POST` for `php://input` for requests where `$_POST` is empty but `php://input` is not. (77cf6130897bc6830bcc053114726ca4f5db20dc)

**Reply by brandonkelly** [reply]
Craft 3.7.17 is out with this change.
